core/domain/stats_services.py

```diff
@@ -173,10 +176,12 @@ def handle_stats_creation_for_new_exploration(exp_id, exp_version, state_names):
     exploration_stats = stats_domain.ExplorationStats.create_default(
         exp_id, exp_version, state_stats_mapping)
     create_stats_model(exploration_stats)
+    return exploration_stats
```

Comment:
Hm -- maybe, for these functions: change the name to "get_" and make a note in the docstring saying clearly "this doesn't save the model, callers should do it" -- then let  the callers do it. Try to avoid default args since they result in more complexity because of the branching.

Reply:
Agreed, seems we should get rid of the `should_put_new_model` entirely.